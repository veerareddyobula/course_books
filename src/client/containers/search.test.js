import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import SearchContainer from './search';

const mockStore = configureStore([]);

describe('SearchContainer Component', () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      fetchBookByTitle: [],
      selectedBooks: () => true,
      booksTitlesList: []
    });
    store.dispatch = jest.fn();
    component = renderer.create(
      <Provider store={store}>
        <SearchContainer />
      </Provider>
    );
  });
  it('SearchContainer Component Snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
