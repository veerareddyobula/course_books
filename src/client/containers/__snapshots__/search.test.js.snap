// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchContainer Component SearchContainer Component Snapshot 1`] = `
Array [
  <h1>
    Search Books
  </h1>,
  <div
    className="search-form"
  >
    <div
      className="auto-complete"
    >
      <input
        onChange={[Function]}
        placeholder="Search books"
        type="text"
        value=""
      />
      
    </div>
    <div>
      <button
        className="btn-primary"
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
    </div>
  </div>,
  <div
    className="mt-2"
  >
    <div>
      No Books Selected
    </div>
  </div>,
]
`;
